lib/seattleflu/db/cli/command/etl/longitudinal.py

```diff
@@ -0,0 +1,303 @@
+
+                if sample:
+                    sample = update_sample(db,
+                        sample = sample,
+                        encounter_id = encounter.id)
+
+                mark_processed(db, record.id, {"status": "processed"})
+
+                LOG.info(f"Finished processing longitudinal record {record.id}")
+
+    except Exception as error:
+        processed_without_error = False
+
+        LOG.error(f"Aborting with error")
+        raise error from None
+
+    else:
+        processed_without_error = True
+
+    finally:
+        if action == "prompt":
+            ask_to_commit = \
+                "Commit all changes?" if processed_without_error else \
+                "Commit successfully processed longitudinal records up to this point?"
+
+            commit = click.confirm(ask_to_commit)
+        else:
+            commit = action == "commit"
+
+        if commit:
+            LOG.info(
+                "Committing all changes" if processed_without_error else \
+                "Committing successfully processed longitudinal records up to this point")
+            db.commit()
+
+        else:
+            LOG.info("Rolling back all changes; the database will not be modified")
+            db.rollback()
+
+
+def site_identifier(site_name: str) -> str:
+    """
+    Given a *site_name*, returns its matching site identifier.
+    """
+    site_map = {
+        1: "HutchKids",
+        2: "WestCampusChildCareCenter"
+    }
+    if site_name not in site_map:
+        raise UnknownSiteError(f"Unknown site name «{site_name}»")
+
+    return site_map[site_name]
+
+
+def encounter_details(document: dict) -> dict:
+    """
+    Describe encounter details in a simple data structure designed to be used
+    from SQL.
+    """
+    return {
+            "age": age_to_delete(document.get("age")), # XXX TODO: Remove age from details
+            "locations": {
+                "home": {
+                    "region": document.get("census_tract"),
+                }
+            },
+            "responses": {
+                "Race": race(document.get("Race")),
+                "FluShot": flu_shot(document.get("FluShot")),
+                "AssignedSex": [sex(document.get("AssignedSex"))],
+                "HispanicLatino": hispanic_latino(document.get("HispanicLatino")),
+                "MedicalInsurance": insurance(document.get("MedicalInsurance"))
+            },
+        }
+
+
+def sex(sex_name) -> str:
+    """
+    Given a *sex_name*, returns its matching sex identifier.
+    """
+    sex_map = {
+        1: "male",
+        2: "female"
+    }
+
+    return sex_map.get(sex_name, "other")
+
+
+def race(races: list) -> list:
+    """
+    Given a *race_name*, returns the matching race identifier found in Audere
+    survey data. Converts multiple races to 'other'.
+    """
+    if races is None:
+        LOG.debug("No race response found.")
+        return [None]
+
+    if len(races) > 1:
+        return 'other'
```

Comment:
I disliked this logic, too. But I wanted to map as closely as possible to `clinical`. There, 'MultipleRaces' gets coded as 'other'.

Reply:
I think `MultipleRaces` → `other` is appropriate because in that case we don't know what they are. But if we have actual multiple race values, then we can map them into the Audere value set.